dotCMS/src/main/java/com/dotcms/webdav/FsDirectoryResource.java

```diff
@@ -0,0 +1,263 @@
+                + "    var L_EditDocumentError_Text = \"The edit feature requires a SharePoint-compatible application and Microsoft Internet Explorer 4.0 or greater.\";\n"
+                + "    var L_EditDocumentRuntimeError_Text = \"Sorry, couldnt open the document.\";\n"
+                + "    function editDocument(strDocument) {\n"
+                + "      if (fNewDoc) {\n"
+                + "        if (!EditDocumentButton.EditDocument(strDocument)) {\n"
+                + "          alert(L_EditDocumentRuntimeError_Text); \n"
+                + "        }\n"
+                + "      } else { \n"
+                + "        alert(L_EditDocumentError_Text); \n"
+                + "      }\n"
+                + "    }\n"
+                + "  </script>\n");
+
+
+
+        w.close("head");
+        w.open("body");
+        w.begin("h1").open().writeText(this.getName()).close();
+        w.open("table");
+        for (Resource r : getChildren()) {
+            w.open("tr");
+
+            w.open("td");
+            String path = buildHref(uri, r.getName());
+            w.begin("a").writeAtt("href", path).open().writeText(r.getName()).close();
+
+            w.begin("a").writeAtt("href", "#").writeAtt("onclick", "editDocument('" + path + "')").open().writeText("(edit with office)").close();
+
+            w.close("td");
+
+            w.begin("td").open().writeText(r.getModifiedDate() + "").close();
+            w.close("tr");
+        }
+        w.close("table");
+        w.close("body");
+        w.close("html");
+        w.flush();
+    }
+
+    @Override
+    public Long getMaxAgeSeconds(Auth auth) {
+        return null;
+    }
+
+    @Override
+    public String getContentType(String accepts) {
+        return "text/html";
+    }
+
+    @Override
+    public Long getContentLength() {
+        return null;
+    }
+
+    private String buildHref(String uri, String name) {
+        /* TODO: Don't know why this has to be the request's absolute URL and the comment
+        doesn't help either. The HttpManager global variables are no longer available in
+        sendContent(), as it might be called after the call left HttpManager.
+
+        // hmm, we're ignoring the path passed in uri. Dodgy...
+        String abUrl = HttpManager.request().getAbsoluteUrl();
+        */
+        String abUrl = uri;
+
+        if (!abUrl.endsWith("/")) {
+            abUrl += "/";
+        }
+        Logger.info(FileSystemResourceFactory.class,"url: " + abUrl);
+        if (ssoPrefix == null) {
+            return abUrl + name;
+        } else {
+            // This is to match up with the prefix set on SimpleSSOSessionProvider in MyCompanyDavServlet
+            String s = insertSsoPrefix(abUrl, ssoPrefix);
+            return s += name;
+        }
+    }
+
+    public static String insertSsoPrefix(String abUrl, String prefix) {
+        // need to insert the ssoPrefix immediately after the host and port
+        int pos = abUrl.indexOf("/", 8);

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Put single-quotes around '/' to use the faster "indexOf(char)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3027)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 8 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)